api/authentication.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime
+import json
+import requests
+
+from django.conf import settings
+from django.core.cache import cache
+
+from allauth.socialaccount.models import SocialAccount
+from rest_framework.authentication import BaseAuthentication
+
+
+def get_cache_key(token):
+    return f'fxa_token_{token}'
+
+
+class FxaTokenAuthentication(BaseAuthentication):
+    def authenticate(self, request):
+        if 'Authorization' not in request.headers:
+            return None
+
+        authorization = request.headers.get('Authorization')
+        if not authorization.startswith('Bearer '):
+            return None
+
+        token = request.headers.get('Authorization').split(' ')[1]
+        cache_key = get_cache_key(token)
+        fxa_resp = cache.get(cache_key)
+        if not fxa_resp:
+            introspect_token_url = (
+                '%s/introspect' %
+                settings.SOCIALACCOUNT_PROVIDERS['fxa']['OAUTH_ENDPOINT']
+            )
+            fxa_resp = requests.post(introspect_token_url, json={'token': token})
+            if not fxa_resp.status_code == 200:
+                # cache anything besides 200 for 60s - it might be an error
+                # we need to re-try, but we don't want to send un-throttled
+                # retries at FXA
+                cache.set(cache_key, fxa_resp, 60)
+                return None
```

Comment:
Same comment as above: I'm doing `return None` on purpose, so that DRF will try to authenticate the request with one of the other authentication backends. (per [docs for custom auth backends](https://www.django-rest-framework.org/api-guide/authentication/#custom-authentication).) The cache is defensive, so that an out-of-control or infinite-looping client doesn't trigger a full FXA request every time - only once ever 60s.